cli/config/configfile/file_test.go

```diff
@@ -481,6 +481,57 @@ func TestLoadFromReaderWithUsernamePassword(t *testing.T) {
 	}
 }
 
+func TestGetAllCredentialsFromEnvironment(t *testing.T) {
+	t.Run("case=can fetch credentials from DOCKER_AUTH_CONFIG and underlying store", func(t *testing.T) {
+		configFile := New("filename")
+		exampleAuth := types.AuthConfig{
+			Username: "user",
+			Password: "pass",
+		}
+		configFile.AuthConfigs["example.com/foo"] = exampleAuth
+
+		envConfig := &ConfigFile{
+			AuthConfigs: map[string]types.AuthConfig{
+				"example.com/foo/env": {
+					Username:      "env_user",
+					Password:      "env_pass",
+					ServerAddress: "example.com/foo/env",
+				},
+			},
+		}
+		data, err := json.Marshal(envConfig)
+		assert.NilError(t, err)
```

Comment:
Is this using the correct / expected marshalling format? i.e., this uses;

```json
{"auths":{"example.com/foo/env":{"username":"env_user","password":"env_pass","serveraddress":"example.com/foo/env"}}}
```

Whereas the format in `~/.docker/config.json` uses the marshalling implemented in `SaveToWriter`, which base64-encodes the content and discards additional fields;

```go
data := bytes.Buffer{}
_ = envConfig.SaveToWriter(&data)
t.Log(data.String())
```

```json
{
     "auths": {
        "example.com/foo/env": {
            "auth": "ZW52X3VzZXI6ZW52X3Bhc3M="
        }
    }
}
```

I think that's the format we expect the env-var to be in 🤔

Reply:
Yes, you're right! I think it could also be assumed to be both. 
Logging in through the CLI would produce the `auth` field, but we don't know if other tools craft a "login" differently.

I'll add a test for both just in case.

Reply:
Done